cmake/Conan.cmake

```diff
@@ -43,18 +38,14 @@ macro(run_conan)
 
     # PATH_OR_REFERENCE ${CMAKE_SOURCE_DIR} is used to tell conan to process
     # the external "conanfile.py" provided with the project
-    # Alternatively a conanfile.txt could be used 
+    # Alternatively a conanfile.txt could be used
     conan_cmake_install(
-      PATH_OR_REFERENCE
-      ${CMAKE_SOURCE_DIR}
-      BUILD
-      missing
+      ERROR_QUIET
```

Comment:
I am unclear on the reason you're adding `ERROR_QUIET` here. As I understand the [docs](https://cmake.org/cmake/help/latest/command/execute_process.html?highlight=error_quiet), `ERROR_QUIET` silences Conan's output to stderr. I think this could make it very difficult to figure out what went wrong. Can you explain your reasoning here?

Reply:
When I started testen, `canon` and `cmake` reported errors I could not understand. 
To analyse it, I added this option.

Reply:
this was the output:

```bash
----Running------
> cd '/Users/clausklein/.conan/data/docopt.cpp/0.6.2/_/_/build/2f6fd32d99347efdec5280d9a5c607e18b4e39e1/build_subfolder' &&
> cmake -G "Ninja" -DCMAKE_BUILD_TYPE="Debug" -DCMAKE_OSX_ARCHITECTURES="x86_64"
> -DCMAKE_OSX_SYSROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk"
> -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="clang" -DCONAN_COMPILER_VERSION="12" -DCONAN_CXX_FLAGS="-m64"
> -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libc++" -DBUILD_SHARED_LIBS="ON"
> -DCMAKE_INSTALL_PREFIX="/Users/clausklein/.conan/data/docopt.cpp/0.6.2/_/_/package/2f6fd32d99347efdec5280d9a5c607e18b4e39e1"
> -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib"
> -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share"
> -DCMAKE_MODULE_PATH="/Users/clausklein/.conan/data/docopt.cpp/0.6.2/_/_/build/2f6fd32d99347efdec5280d9a5c607e18b4e39e1"
> -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DUSE_BOOST_REGEX="False" -Wno-dev
> '/Users/clausklein/.conan/data/docopt.cpp/0.6.2/_/_/build/2f6fd32d99347efdec5280d9a5c607e18b4e39e1'
-----------------
Logging command output to file '/Users/clausklein/.conan/data/docopt.cpp/0.6.2/_/_/build/2f6fd32d99347efdec5280d9a5c607e18b4e39e1/conan_run.log'
CMake Error at /usr/local/Cellar/cmake/3.21.3/share/cmake/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  /usr/local/bin/gcc-10.
Call Stack (most recent call first):
  CMakeLists.txt:2 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!

```

Reply:
If `conan_install` failed, should not CMake fail as well? In this case `ERROR_QUIET` allows CMake to continue when it should probably fail.

If you are able to get this project running at all, and the Conan process is failing, you must have installed the dependencies independently of Conan. If this is the case, you can turn off Conan with the `ENABLE_CONAN` CMake option.

The `Could not find compiler set in environment variable CC` looks like a configuration problem; it is not the expected result, and will not result in a properly built binary.

Reply:
To remove noise add `OUTPUT_QUIET`. `ERROR_QUIET` should not be used as it hides the errors

Reply:
ok